cellar_rebalancer/src/allocation.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{cellars::uniswapv3::UniswapV3CellarTickInfo, prelude::*, somm_send};
+use abscissa_core::Application;
+use deep_space::{Coin, Contact};
+use ethers::prelude::U256;
+use proto::query_client::QueryClient as AllocationQueryClient;
+use somm_proto::somm as proto;
+use std::time::Duration;
+use tokio::time::{sleep, timeout};
+use tonic::transport::Channel;
+
+pub struct Connections {
```

Comment:
We should try to resolve whether we successfully created clients or not at the time of instantiating them, this way we can pass a `Connections` struct of clients rather than `Options`.